src/perf/event_provider.cpp

```diff
@@ -345,6 +346,18 @@ static void event_description_update(CounterDescription& event, std::uint64_t va
     }
 }
 
+const CounterDescription raw_read_event(const std::string& ev_desc)
+{
+    uint64_t code = std::stoull(ev_desc.substr(1), nullptr, 16);
+    const CounterDescription event(ev_desc, PERF_TYPE_RAW, code, 0);
+
+    if (!event_is_openable(event))
+    {
+        throw EventProvider::InvalidEvent("cannot open requested event");
```

Comment:
Add the name of the event, which failed

Reply:
Is that needed? The error message in that case ( Example: " 'r003c' does not name a known event, ignoring! (reason: cannot open requested event)") already contains the name of the event

Reply:
Oh, it get's added somewhere higher in call stack?

Reply:
Seems like that